### PR TITLE
Add org.virt_manager.virt-manager

### PR DESCRIPTION
**Pull request by travier**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly:
    - Manage Virtual Machine using libvirt
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed:
    - See: https://virt-manager.org/ & https://flathub.org/apps/org.virt_manager.virt-viewer
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally
    - It works for me so far in what I've tested.
- [ ] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission:
    - I've contacted upstream authors. See: https://github.com/virt-manager/virt-manager/issues/246#issuecomment-2497399765

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by travier** [comment]
bot, build org.virt_manager.virt-manager

File: org.virt_manager.virt-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+                  stable-only: true
+                  url-template: https://gitlab.freedesktop.org/spice/spice-protocol/-/archive/v$version/spice-protocol-v$version.tar.gz
+
+      - name: python-gobject
+        buildsystem: meson
+        config-opts:
+          - -Dpycairo=enabled
+        sources:
+          - type: archive
+            url: https://download.gnome.org/sources/pygobject/3.50/pygobject-3.50.0.tar.xz
```
Comment (bbhtt) [inline org.virt_manager.virt-manager.yaml:376]: should be in the runtime

**Comment by bbhtt** [comment]
Icon gone from about window.

image

**Comment by bbhtt** [comment]
How well does this work inside Flatpak?

**Comment by bbhtt** [comment]
Qemu/kvm user session option is missing things.

image

**Comment by travier** [comment]
Yes, this is a libvirt client GUI only, you will need libvirtd installed and running on your host, either at the system or session level.

File: virt-manager-systray-icon-use-reverse-dns-naming-convention.patch
Comment (hfiguiere) [inline virt-manager-systray-icon-use-reverse-dns-naming-convention.patch:1]: if you add `copy-icon: true` next to `rename-icon` then this patch shouldn't be necessary.

File: metadata.patch [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+--- a/data/virt-manager.appdata.xml.in
++++ b/data/virt-manager.appdata.xml.in
+@@ -1,8 +1,11 @@
+ <?xml version="1.0" encoding="UTF-8"?>
+ <component type="desktop">
+-  <id>virt-manager.desktop</id>
++  <id>org.virt_manager.virt-manager</id>
+   <metadata_license>CC0-1.0</metadata_license>
+   <project_license>GPL-2.0+</project_license>
++  <developer id="org.virt-manager">
```
Comment (bbhtt) [inline metadata.patch:21]:
```suggestion
+  <developer id="org.virt_manager">
```

File: metadata.patch [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+   <summary>Graphically manage KVM, Xen, or LXC via libvirt</summary>
+   <description>
+@@ -40,6 +43,10 @@
+     <keyword>kvm</keyword>
+   </keywords>
+   <content_rating type="oars-1.1"/>
++  <launchable type="desktop-id">org.virt_manager.virt-manager.desktop</launchable>
++  <replaces>
++    <id>virt-manager.desktop</id>
++  </replaces>
```
Comment (bbhtt) [inline metadata.patch:34]: this should be under `provides` tag.

File: virt-manager-systray-icon-use-reverse-dns-naming-convention.patch
Comment (travier) [reply virt-manager-systray-icon-use-reverse-dns-naming-convention.patch:1]: I think we should upstream this patch. Will start a discussion upstream.

File: virt-manager-disable-man-pages.patch [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+-            "man/virt-install.1",
+-            "man/virt-clone.1",
+-            "man/virt-xml.1"
+-        ]),
++        #("share/man/man1", [
++        #    "man/virt-manager.1",
++        #    "man/virt-install.1",
++        #    "man/virt-clone.1",
++        #    "man/virt-xml.1"
++        #]),
```
[the remark this answers is not in the document]
Comment (travier) [reply virt-manager-disable-man-pages.patch:38]: Will test that.

[the remark this answers is not in the document]
Comment (travier) [reply virt-manager-disable-man-pages.patch]: Removed.

**Comment by travier** [comment]
bot, build org.virt_manager.virt-manager

File: virt-manager-disable-man-pages.patch (hunk 4 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply virt-manager-disable-man-pages.patch]: you can add cleanup rules to delete the man pages. that's what we do usually as they are not accessible.

File: org.virt_manager.virt-manager.yaml [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+  - --filesystem=xdg-pictures
+  - --filesystem=xdg-public-share
+  - --filesystem=xdg-videos
+  # Access to the UNIX socket for the local libvirtd system instance
+  - --filesystem=/run/libvirt
+  # Access to ssh configuration and keys
+  - --filesystem=~/.ssh
+  # Access to ssh-agent pid and socket
+  - --socket=ssh-auth
+  - --own-name=org.virt-manager.virt-manager
```
Comment (bbhtt) [inline org.virt_manager.virt-manager.yaml:26]: this should be fixed upstream

Comment (travier) [reply org.virt_manager.virt-manager.yaml:26]: Agree, will do.

**Pull request closed without merging**